### PR TITLE
terraform: add save to file feature to apikeys resource #5062

### PR DESCRIPTION
**Pull request by pooraniss-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by pooraniss-ibm** [comment]
Screenshot 2021-06-23 at 15 28 54
Screenshot 2021-06-23 at 15 29 17

File: ibm/data_source_ibm_iam_api_key.go [hunk 1]
```diff
@@ -131,9 +131,6 @@ func dataSourceIbmIamApiKeyRead(context context.Context, d *schema.ResourceData,
 	if err = d.Set("account_id", apiKey.AccountID); err != nil {
 		return diag.FromErr(fmt.Errorf("Error setting account_id: %s", err))
 	}
-	if err = d.Set("apikey", apiKey.Apikey); err != nil {
-		return diag.FromErr(fmt.Errorf("Error setting apikey: %s", err))
```
Comment (hkantare) [inline ibm/data_source_ibm_iam_api_key.go:135]: Remove the attribute from schema also

File: ibm/resource_ibm_iam_api_key.go [hunk 2]
```diff
@@ -142,7 +151,7 @@ func resourceIbmIamApiKeyCreate(context context.Context, d *schema.ResourceData,
 	d.Set("apikey", *apiKey.Apikey)
 
 	if keyfile, ok := d.GetOk("file"); ok {
-		if err := saveToFile(apiKey, keyfile.(string)); err != nil {
+		if err := saveApikeyToFile(apiKey, keyfile.(string)); err != nil {
```
Comment (hkantare) [inline ibm/resource_ibm_iam_api_key.go:154]: you can use same saveToFile method why we need a seperate method ?

File: ibm/resource_ibm_iam_api_key.go [hunk 3]
```diff
@@ -170,44 +179,41 @@ func resourceIbmIamApiKeyRead(context context.Context, d *schema.ResourceData, m
 		return diag.FromErr(err)
 	}
```
Comment (hkantare) [inline ibm/resource_ibm_iam_api_key.go:181]: In read method here in below code add apikey == nil condition also
apiKey, response, err := iamIdentityClient.GetAPIKey(getApiKeyOptions)
	if err != nil || apikey == nil {

File: website/docs/r/iam_api_key.html.markdown [hunk 4]
```diff
@@ -28,6 +28,7 @@ Review the argument references that you can specify for your resource.
 - `entity_lock` - (Optional, Bool) Indicates the API key is locked for further write operations. Default value is `false`.
 - `name` - (Required, String) The name of the API key. The name is not checked for uniqueness. Therefore, multiple names with the same value can exist. Access is done through the UUID of the API key.
 - `store_value` - (Optional, Bool) Use `true` or `false` to set whether the API key value is retrievable in the future by using the `Get` details of an API key request. If you create an API key for a user, you must specify `false` or omit the value. Users cannot store the API key.
+- `file` - (Optional, String) The file name where API key is to be stored.
```
Comment (hkantare) [inline website/docs/r/iam_api_key.html.markdown:33]: Remove the apiKey from datasource doc

**Pull request merged**